### PR TITLE
Adding a Done option in the deadline adapter

### PR DESCRIPTION
**Pull request by Matathiasq13824**
This PR add the option to change the state of the deadline in the `MainActivity` in addition of `DeadlineDetailsActivity` to facilitate the usage of the app.
Adding a `ToggleButton` in the items of the `listView` was strangly much more complicated than expected and some strange behaviors had appeared but were fixed.

### CODE REVIEW

File: app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt [hunk 1]
```diff
@@ -108,15 +115,34 @@ class DeadlineAdapter(private val context: Context) : BaseAdapter() {
...
+        updateDetails(detailTextView, deadline)
+
+        // Set the checkbox
+        doneCheckbox.isChecked = (deadline.state == DeadlineState.DONE)
+        doneCheckbox.setOnCheckedChangeListener { _, isChecked ->
+            val newDeadline = Deadline(
+                deadline.title,
+                if (isChecked) DeadlineState.DONE else DeadlineState.TODO,
+                deadline.dateTime
+            )
```
Comment (felko) [inline app/src/main/java/com/github/multimatum_team/multimatum/adaptater/DeadlineAdapter.kt:127]: You can use `deadline.copy(state = if (isChecked) DeadlineState.DONE else DeadlineState.TODO)` instead (kotlin data classes automatically generates the `copy` method)

**Review by felko — APPROVED** [review]

File: app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt [hunk 2]
```diff
@@ -79,16 +77,13 @@ class MainActivity : AppCompatActivity() {
...
     private fun setDeleteOnSweep(lv: ListView, viewModel: DeadlineListViewModel) {
-
         // Create a Listener who will delete the given deadline if swept
         val touchListener = SwipeToDismissTouchListener(
             ListViewAdapter(lv),
             object : SwipeToDismissTouchListener.DismissCallbacks<ListViewAdapter?> {
                 override fun canDismiss(position: Int): Boolean {
-                    return true
-                }
+                    return true }
```
Comment (ValentinAebi) [inline app/src/main/java/com/github/multimatum_team/multimatum/activity/MainActivity.kt:86]: The `}` on the same line as the `return` looks like a typo

File: app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt [hunk 3]
```diff
@@ -209,26 +224,66 @@ class DeadlineAdapterTest {
...
         Assert.assertEquals(
             Typeface.ITALIC,
             (listItemView.findViewById<TextView>(R.id.deadline_list_subtitle)).typeface.style
         )
         //check details
+        val deadlineListDetails = listItemView.findViewById<TextView>(R.id.deadline_list_detail)
         Assert.assertEquals(
             "Is already Due",
-            (listItemView.findViewById<TextView>(R.id.deadline_list_detail)).text
+            (deadlineListDetails).text
```
Comment (ValentinAebi) [inline app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt:238]: I think parentheses are not needed here

File: app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt [hunk 4]
```diff
@@ -209,26 +224,66 @@ class DeadlineAdapterTest {
...
+
+    @Test
+    fun `The button in the item change the state of the deadline`() {
+        val parent = ListView(context)
+        val listItemView: View = adapter.getView(0, null, parent)
+        //check details before
+        val deadlineListDetails = listItemView.findViewById<TextView>(R.id.deadline_list_detail)
+        Assert.assertEquals(
+            "Is already Due",
+            (deadlineListDetails).text
```
Comment (ValentinAebi) [inline app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt:259]: Are these parentheses really necessary?

File: app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt [hunk 5]
```diff
@@ -209,26 +224,66 @@ class DeadlineAdapterTest {
...
+        )
+        Assert.assertEquals(
+            Typeface.NORMAL,
+            (deadlineListDetails)
```
Comment (ValentinAebi) [inline app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt:263]: Parentheses here too (?)

File: app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt [hunk 6]
```diff
@@ -209,26 +224,66 @@ class DeadlineAdapterTest {
...
+                .typeface.style
         )
         Assert.assertEquals(
             Color.BLACK,
-            (listItemView.findViewById<TextView>(R.id.deadline_list_detail).currentTextColor)
+            (deadlineListDetails.currentTextColor)
```
Comment (ValentinAebi) [inline app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt:268]: Same comment about parentheses...

File: app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt [hunk 7]
```diff
@@ -209,26 +224,66 @@ class DeadlineAdapterTest {
...
+        )
+        //Set in done
+        listItemView.findViewById<ToggleButton>(R.id.deadline_list_check_done).performClick()
+        //check details after
+        Assert.assertEquals(
+            "Done",
+            (deadlineListDetails).text
```
Comment (ValentinAebi) [inline app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt:275]: Here too

File: app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt [hunk 8]
```diff
@@ -209,26 +224,66 @@ class DeadlineAdapterTest {
...
         )
+        Assert.assertEquals(
+            Typeface.BOLD,
+            (deadlineListDetails)
```
Comment (ValentinAebi) [inline app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt:279]: Parentheses?

File: app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt [hunk 9]
```diff
@@ -209,26 +224,66 @@ class DeadlineAdapterTest {
...
+                .typeface.style
+        )
+        Assert.assertEquals(
+            Color.GREEN,
+            (deadlineListDetails.currentTextColor)
```
Comment (ValentinAebi) [inline app/src/test/java/com/github/multimatum_team/multimatum/DeadlineAdapterTest.kt:284]: Parentheses?

**Review by ValentinAebi — APPROVED** [review]

**Pull request merged**